src/plugins/ui_core_views/cell_evaluation/evaluator.ts

```diff
@@ -500,6 +509,9 @@ export class Evaluator {
         this.getters.getLocale(),
         cell
       );
+      if (!evaluatedCell.origin) {
+        evaluatedCell.origin = position;
```

Comment:
an alternative approach to all this would be to add the origin when accessing the reference value in `refFn` and `range`.
It means a cell with a formula which is not returning a reference wouldn't have a `.origin` (nor literal cells), but that makes (more?) sense.

I would only be 2 places to change instead of the 5 or more here.